pySDC/tests/test_sweepers/test_MPI_sweeper.py

```diff
@@ -171,15 +189,36 @@ def test_sweeper_NCCL(num_nodes, quad_type, residual_type, imex, initGuess, laun
         imex (bool): Use IMEX sweeper or not
         launch (bool): If yes, it will launch `mpirun` with the required number of processes
     """
-    individual_test(num_nodes, quad_type, residual_type, imex, initGuess, useNCCL=True, launch=launch)
+    individual_test(
+        num_nodes=num_nodes,
+        quad_type=quad_type,
+        residual_type=residual_type,
+        imex=imex,
+        init_guess=init_guess,
+        useNCCL=True,
+        ML=False,
+        launch=launch,
+    )
 
 
 if __name__ == '__main__':
-    import sys
-
-    if '--test_sweeper' in sys.argv:
-        imex = False if sys.argv[-3] == 'False' else True
-        useNCCL = False if sys.argv[-1] == 'False' else True
-        individual_test(
-            sys.argv[-6], sys.argv[-5], sys.argv[-4], imex=imex, initGuess=sys.argv[-2], useNCCL=useNCCL, launch=False
-        )
+    str_to_bool = lambda me: False if me == 'False' else True
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--ML', type=str_to_bool, help='Multilevel toogle', choices=[True, False])
+    parser.add_argument('--num_nodes', type=int, help='Number of collocation nodes')
+    parser.add_argument('--quad_type', type=str, help='Quadrature rule', choices=['GAUSS', 'RADAU-RIGHT', 'RADAU-LEFT'])
+    parser.add_argument(
+        '--residual_type',
+        type=str,
+        help='Way of computing the residual',
+        choices=['full_rel', 'last_abs', 'full_abs', 'last_rel'],
+    )
+    parser.add_argument('--imex', type=str_to_bool, help='Toggle for IMEX', choices=[True, False])
+    parser.add_argument('--useNCCL', type=str_to_bool, help='Toggle for NCCL communicator', choices=[True, False])
+    parser.add_argument('--init_guess', type=str, help='Initial guess', choices=['spread', 'copy', 'zero'])
+    args = parser.parse_args()
+
+    kwargs = {me[0]: me[1] for me in args._get_kwargs()}
```

Comment:
Simpler like this :

```python
kwargs = vars(args)
```
which you even more simplify in the following line :

```python
individual_test(**vars(args))
```

Reply:
Hope you didn't spend too much time finding this other solution :laughing: 

![image](https://github.com/Parallel-in-Time/pySDC/assets/12183839/c3ce4d7a-a9ce-4c6f-a320-470fa2e5ce22)

Reply:
Thanks, that's indeed nicer! I was just going from the argparse docs which didn't take too long. Maybe I should use chatGPT more...